barretenberg/cpp/src/barretenberg/commitment_schemes/ipa/ipa.test.cpp

```diff
@@ -330,3 +330,84 @@ TEST_F(IPATest, ShpleminiIPAWithShift)
 
     EXPECT_EQ(result, true);
 }
+/**
+ * @brief Test the behaviour of the method ShpleminiVerifier::remove_shifted_commitments
+ *
+ */
+TEST_F(IPATest, ShpleminiIPAShiftsRemoval)
+{
+    using IPA = IPA<Curve>;
+    using ShplonkProver = ShplonkProver_<Curve>;
+    using ShpleminiVerifier = ShpleminiVerifier_<Curve>;
+    using GeminiProver = GeminiProver_<Curve>;
+
+    const size_t n = 8;
+    const size_t log_n = 3;
+
+    // Generate multilinear polynomials, their commitments (genuine and mocked) and evaluations (genuine) at a random
+    // point.
+    auto mle_opening_point = this->random_evaluation_point(log_n); // sometimes denoted 'u'
```

Comment:
We really need to clean up these pcs test files by having single methods that generate input rather than duplicating code over and over again  (similar to what we do for circuits). I previously added an issue on this. Just saying, not suggesting that it's something that should be done in this PR